jwst/tweakreg/tests/test_tweakreg.py

```diff
@@ -82,11 +94,229 @@ def test_write_catalog(dummy_source_catalog, tmp_cwd):
     '''
 
     OUTDIR = 'outdir'
-    model = ImageModel()
     step = tweakreg_step.TweakRegStep()
     os.mkdir(OUTDIR)
     step.output_dir = OUTDIR
     expected_outfile = os.path.join(OUTDIR, 'catalog.ecsv')
-    step._write_catalog(model, dummy_source_catalog, 'catalog.ecsv')
+    step._write_catalog(dummy_source_catalog, 'catalog.ecsv')
+
+    assert os.path.exists(expected_outfile)
+
+
+@pytest.fixture()
+def example_wcs():
+    path = os.path.join(
+        os.path.dirname(__file__),
+        "data",
+        "nrcb1-wcs.asdf")
+    with asdf.open(path, lazy_load=False) as af:
+        return af.tree["wcs"]
+
+
+@pytest.fixture()
+def example_input(example_wcs):
+    m0 = ImageModel((512, 512))
+
+    # add a wcs and wcsinfo
+    m0.meta.wcs = example_wcs
+    m0.meta.wcsinfo = _wcsinfo_from_wcs_transform(example_wcs)
+
+    # and a few 'sources'
+    m0.data[:] = 0.001
+    n_sources = N_EXAMPLE_SOURCES  # a few more than default minobj
+    rng = np.random.default_rng(26)
+    xs = rng.choice(50, n_sources, replace=False) * 8 + 10
+    ys = rng.choice(50, n_sources, replace=False) * 8 + 10
+    for y, x in zip(ys, xs):
+        m0.data[y-1:y+2, x-1:x+2] = [
+            [0.1, 0.6, 0.1],
+            [0.6, 0.8, 0.6],
+            [0.1, 0.6, 0.1],
+        ]
+
+    m1 = m0.copy()
+    # give each a unique filename
+    m0.meta.filename = 'some_file_0.fits'
+    m1.meta.filename = 'some_file_1.fits'
+    c = ModelContainer([m0, m1])
+    return c
+
+
+def test_tweakreg_step(example_input):
+    # shift 9 pixels
+    example_input[1].data = np.roll(example_input[1].data, 9, axis=0)
+
+    # assign images to different groups (so they are aligned to each other)
+    example_input[0].meta.group_id = 'a'
+    example_input[1].meta.group_id = 'b'
+    step = tweakreg_step.TweakRegStep()
+    result = step(example_input)
+
+    # check that step completed
+    for model in result:
+        assert model.meta.cal_step.tweakreg == 'COMPLETE'
+
+    # and that the wcses are different
+    abs_delta = abs(result[1].meta.wcs(0, 0)[0] - result[0].meta.wcs(0, 0)[0])
+    assert abs_delta > 1E-5
```

Comment:
Thanks for taking a look!

I added some comments and a short docstring in: https://github.com/spacetelescope/jwst/pull/8424/commits/d4131909c253049079d04f82aee0fcf128408eef

Does that help to explain what the test is checking?

I did not find a unit test for `TweakRegStep` (is there one that already exists?).

Reply:
I picked 9 here since none of the fake sources are at the edges. That dependence is not obvious in the tests so I replaced this with a shift using slicing that also sets the other pixels to the background level (see: f9a27a9ce).

I also parametrized this test to run without the shift as you mentioned. The outputs are different with and without the shift (as expected).